apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/EnumMap.cs

```diff
@@ -71,6 +74,41 @@ internal static string ToApiValue(T value, string paramName = "value")
             throw new ArgumentException($"Value {value} is undefined in {typeof(T).Name}", paramName);
         }
 
+        /// <summary>
+        /// Similar to <see cref="ToApiValue(T, string)"/> but for used with Flags enum.
+        /// </summary>
+        /// <param name="value">An enum value, possibly flagged.</param>
+        /// <returns>A set of string that contains the api value corresponding to each
+        /// of the flags set in <paramref name="value"/>.</returns>
+        internal static ISet<string> ToApiValues(T value)
+        {
+            Enum eValue = value as Enum;
```

Comment:
I think we can do all of this without any calls to `Enum.GetValues` etc after the type initialization.
It's probably simplest if we get the code merged as-is, then I'll do a follow-up PR. (I know what I'm thinking of, but it's harder to describe than just to do :)

Reply:
I think I understood what you described on the previous PR using an extra dictionary of int to string. This was just a faster change :).
Let me try your proposal and if I get it wrong you can do the follow up PR.